table_schema_cache.go

```diff
@@ -211,6 +211,77 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 	return tableSchemaCache, nil
 }
 
+func (ts *TableSchema) findColumnByName(name string) *schema.TableColumn {
+	for _, column := range ts.Columns {
+		if column.Name == name {
+			return &column
+		}
+	}
+	return nil
+}
+
+func NonExistingPaginationKeyColumnError(schema, table, paginationKey string) error {
+	return fmt.Errorf("Pagination Key `%s` for %s.%s non existent", paginationKey, schema, table)
+}
+
+func NonExistingPKError(schema, table string) error {
+	return fmt.Errorf("%s.%s has no Primary Key to default to for Pagination purposes. Kindly specify a Pagination Key for this table in the CascadingPaginationColumnConfig", schema, table)
+}
+
+func CompositePKError(schema, table string) error {
+	return fmt.Errorf("%s.%s attempted to default to a Composite Primary Key for Pagination purposes whereas this is not supported. Kindly specify a Pagination Key for this table in the CascadingPaginationColumnConfig", schema, table)
+}
+
+func (ts *TableSchema) paginationKeyColumn(dbname, table string, cascadingPaginationColumnConfig *CascadingPaginationColumnConfig) (*schema.TableColumn, error) {
+	var err error
+	var paginationKeyColumn *schema.TableColumn
+
+	if paginationColumn, found := cascadingPaginationColumnConfig.PaginationColumnFor(dbname, table); found {
+		paginationKeyColumn = ts.findColumnByName(paginationColumn)
+		err = NonExistingPaginationKeyColumnError(dbname, table, paginationColumn)
+	} else {
+		if cascadingPaginationColumnConfig == nil || cascadingPaginationColumnConfig.FallbackColumn == "" {
+			switch len(ts.PKColumns) {
+			case 0:
+				err = NonExistingPKError(dbname, table)
+			case 1:
+				paginationKeyColumn = &ts.Columns[ts.PKColumns[0]]
+			default:
+				err = CompositePKError(dbname, table)
+			}
+		} else {
+			paginationKeyColumn = ts.findColumnByName(cascadingPaginationColumnConfig.FallbackColumn)
+			err = NonExistingPaginationKeyColumnError(dbname, table, cascadingPaginationColumnConfig.FallbackColumn)
+		}
+	}
+
+	if paginationKeyColumn == nil {
```

Comment:
`if err != nil { return err }`?

Reply:
Actually you could likely get rid of this whole block and just `return paginationKeyColumn, err`

Reply:
I currently need it because I am always populating `err`

Reply:
I feel like this style is not very common in Go. Looking at the `findColumnByName` method and how it is invoked, it seems to me that method should return a (tablecol, err) tuple, where err == nil if a column can be found.